src/components/ImmatriculationHistory.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+    <div v-if="!busy">
+        <div v-if="chronologicalList.length > 0">
+            <div v-for="(pair, index) in chronologicalList" :key="pair">
+                <immatriculation-history-entry
+                    class="w-1/2"
+                    :is-first-row="index == 0"
+                    :is-last-row="index == chronologicalList.length - 1"
+                    :fields-of-study="pair.fieldsOfStudy"
+                    :semester="pair.semester"
+                />
+            </div>
+        </div>
+        <div v-else>
+            <label class="text-lg">There is no matriculation data, yet!</label>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+    import MatriculationData from "@/api/api_models/matriculation_management/MatriculationData";
+    import { reactive, ref, onBeforeMount } from "vue";
+    import SubjectMatriculation from "@/api/api_models/matriculation_management/SubjectMatriculation";
+    import { FieldOfStudy } from "@/api/api_models/user_management/FieldOfStudy";
+    import { historyToSortedList } from "@/use/ImmatriculationHistoryHandler";
+    import ImmatriculationHistoryEntry from "@/components/ImmatriculationHistoryEntry.vue";
+    import MatriculationManagement from "@/api/MatriculationManagement";
+    import GenericResponseHandler from "@/use/GenericResponseHandler";
+    export default {
+        components: {
+            ImmatriculationHistoryEntry,
+        },
+        props: {
+            username: {
+                type: String,
+                required: true,
+            },
+            busy: {
+                type: Boolean,
+                required: true,
+            },
+        },
+        emits: ["update:busy"],
+        setup(props: any, { emit }: any) {
+            //TODO Remove mock data as soon as the endpoint in backend provides data
+            let history: MatriculationData = reactive({
+                matriculationId: "egal",
+                firstName: "egal",
+                lastName: "egal",
+                birthDate: "egal",
+                matriculationStatus: [
+                    {
+                        fieldOfStudy: "Computer Science" as FieldOfStudy,
+                        semesters: ["WS2020/21", "WS2019/20", "SS2020"],
+                    } as SubjectMatriculation,
+                    {
+                        fieldOfStudy: "Mathematic" as FieldOfStudy,
```

Comment:
```suggestion
                        fieldOfStudy: "Mathematics" as FieldOfStudy,
```
how does this not throw a type error? I hade enums